ocm/__init__.py

```diff
@@ -459,6 +471,42 @@ class ComponentReference(Artifact, LabelMethodsMixin):
     extraIdentity: dict[str, str] = dataclasses.field(default_factory=dict)
     labels: list[Label] = dataclasses.field(default_factory=tuple)
 
+    def normalise(
+        self,
+        component_descriptor_lookup: collections.abc.Callable[[ComponentIdentity], 'ComponentDescriptor'], # noqa: E501
+        image_ref_to_digest_lookup: collections.abc.Callable[[str], str],
+    ) -> list[dict]:
+        raw = dataclasses.asdict(self)
+
+        if labels := [l.normalise() for l in self.labels if l.signing]:
+            raw['labels'] = labels
+        else:
+            del raw['labels']
+
+        if not self.digest:
+            component_descriptor = component_descriptor_lookup(ComponentIdentity(
+                name=self.componentName,
+                version=self.version,
+            ))
+
+            digest = component_descriptor.digest(
+                component_descriptor_lookup=component_descriptor_lookup,
+                image_ref_to_digest_lookup=image_ref_to_digest_lookup,
+            )
+
+            raw['digest'] = dataclasses.asdict(DigestSpec(
+                hashAlgorithm='SHA-256',
+                normalisationAlgorithm=NormalisationAlgorithm.JSON_NORMALISATION,
```

Comment:
should we make the normalisation-algorithm a parameter? we could define the (now) only supported value as default-value.
same for hashAlgorithm, I suppose